File: lab1/src/objects/SortMyVector.java
package objects;
import java.util.Comparator;

import kitpo.IUserType;

import java.util.Comparator;

public class SortMyVector implements Comparator<MyVector> {
    @Override
    public int compare(MyVector o1, MyVector o2) {
        return new Double(o1.getLength()).compareTo(o2.getLength());
    }
}

File: lab3/src/main/kotlin/Objects/SortMyVector.kt
package Objects

import List.IUserType


class SortMyVector : Comparator<IUserType> {
    override fun compare(o1: IUserType, o2: IUserType): Int {
        o1 as MyVector
        o2 as MyVector
        return o1.length.compareTo(o2.length)
    }
}